rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -672,6 +677,16 @@ rmw_node_t * rmw_create_node(const char * name, size_t domain_id)
   }
   memcpy(const_cast<char *>(node_handle->name), name, strlen(name) + 1);
 
+  node_handle->namespace_ =
+    static_cast<const char *>(malloc(sizeof(char) * strlen(namespace_) + 1));
+  if (!node_handle->namespace_) {
+    RMW_SET_ERROR_MSG("failed to allocate memory");
+    free(const_cast<char *>(node_handle->name));
```

Comment:
Also, this is not a bug, it is just undoing `->name` since we need to exit here. Since `->name` was allocated with `malloc` I'll continue to use `free` here.